### PR TITLE
Clean up old test threads

### PR DESCRIPTION
**Pull request by tinwelint**
which was left hanging around until JVM ended.
Even found some places in product which forgot to close things.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/util/DebugUtil.java [hunk 1]
```diff
@@ -111,9 +111,26 @@ public static boolean currentStackTraceContains( Predicate<StackTraceElement> pr
...
+            {
+                continue;
+            }
+            if ( excludeCallingThread && thread == Thread.currentThread() )
+            {
+                continue;
+            }
+
+            out.println( new CallStack( stack.getValue(), "Thread: " + stack.getKey() ) );
+            out.println();
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/util/DebugUtil.java]: We have something similar in `ThreadTestUtil`.

**Comment by chrisvest** [comment]
The build failed to download some dependencies, so I've restarted it.

**Comment by chrisvest** [comment]
@tinwelint When removing `DebugUtil#dumpThreads` you missed a usage site at `BackupServiceStressTestingBuilder.java:[238,30]`

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java [hunk 2]
```diff
@@ -122,7 +122,18 @@ public BatchingNeoStores( FileSystemAbstraction fileSystem, File storeDir, Recor
...
+            IllegalStateException ise =
+                    new IllegalStateException( storeDir + " already contains data, cannot do import here" );
+            try
+            {
+                pageCache.close();
+            }
+            catch ( IOException e )
+            {
+                // Oddly enough we can't close the page cache, how to communicate this? Here we add as suppressed
+                ise.addSuppressed( e );
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java:134]: Yeah, I think that's the right way to communicate this.

**Comment by chrisvest** [comment]
Looks pretty good, otherwise.

**Comment by tinwelint** [comment]
@chrisvest I (actually force-) pushed that one-line fix now

**Comment by tinwelint** [comment]
@chrisvest could potentially be ready to be merged now

**Pull request merged**